staging/src/k8s.io/apimachinery/pkg/fields/selector.go

```diff
@@ -116,6 +116,9 @@ func (t *hasTerm) Requirements() Requirements {
 }
 
 func (t *hasTerm) String() string {
+	if t == nil {
+		return "<nil>"
+	}
```

Comment:
is this used in the serialized selector form? if so, `<nil>` is not valid selector syntax. same for the notHasTerm below

Reply:
Use empty string instead of "<nil>". Please review it again. Thanks

Reply:
Add test for hasTerm/notHasTerm String method when receiver is nil.
TestNilSelectorToString ensure the String method return empty string. And the empty parse is tested in https://github.com/kubernetes/kubernetes/blob/512b7b75a8da19e669e058bf13d5978560c4c65f/staging/src/k8s.io/apimachinery/pkg/fields/selector_test.go#L136